src/main/frontend/src/components/aggregates/AggregateDialog.svelte

```diff
@@ -18,94 +12,83 @@
 
 	export let dialogActive;
 	export let editMode;
+	export let oldAggregate;
+	let newAggregate;
+	let aggregateName, stateFields, events, methods, rootPath, producerExchangeName, consumerExchangeName, schemaGroup, disableSchemaGroup, routes, outgoingEvents, receivers;
 
-	export let currentId;
-	
-	let aggregateIndex = undefined;
-	let aggregateName = $currentAggregate ? $currentAggregate.aggregateName : "";
-	let stateFields = [{ name: "id", type: "String" }];
-	let events = [];
-	let methods = [];
-	let rootPath = "/";
-	let producerExchangeName = "";
-	let consumerExchangeName = "";
-	let schemaGroup = "";
-	let disableSchemaGroup = false;
-	let routes = [];
-	let outgoingEvents = [];
-	let receivers = [];
-
-	const add = () => {
-		if(requireRule(aggregateName)) return;
-		$aggregateSettings = [...$aggregateSettings, $currentAggregate];
-		currentId = undefined;
-		reset();
-		dialogActive = false;
-	}
 
-	const update = () => {
-		if(requireRule(aggregateName)) return;
-		$aggregateSettings.splice(currentId, 1, $currentAggregate);
-		$aggregateSettings = $aggregateSettings;
-		currentId = undefined;
-		reset();
-		dialogActive = false;
+	const retrieveSchemaGroup = () => $aggregateSettings.length > 0 ? $aggregateSettings[0].producerExchange.schemaGroup : "";
+	const canWriteSchemaGroup = () => (schemaGroup == undefined || schemaGroup.length == 0); //currentId == 0 ||
+	const initialAggregate = {
+		aggregateName: "",
+		stateFields: [{ name: "id", type: "String" }],
+		events: [],
+		methods: [],
+		api: {
+			rootPath: "/",
+			routes: [],
+		},
+		producerExchange: {
+			exchangeName: "",
+			schemaGroup: retrieveSchemaGroup(),
+			outgoingEvents: [],
+		},
+		consumerExchange: {
+			exchangeName: "",
+			receivers: [],
+		},
+		disableSchemaGroup: !canWriteSchemaGroup(),
 	}
 
-	const reset = () => {
-		aggregateName = "";
-		stateFields = [{ name: "id", type: "String" }];
-		events = [];
-		methods = [];
-		rootPath = "/";
-		routes = [];
-		producerExchangeName = "";
-		schemaGroup = retrieveSchemaGroup();
-		disableSchemaGroup = !canWriteSchemaGroup();
-		outgoingEvents = [];
-		consumerExchangeName = "";
-		receivers = [];
+	const add = () => {
+		if(!valid) return;
+		$aggregateSettings = [...$aggregateSettings, newAggregate];
+		closeDialog();
 	}
 
-	const retrieveSchemaGroup = () => {
-		return $aggregateSettings.length > 0 ? $aggregateSettings[0].producerExchange.schemaGroup : "";
+	const update = () => {
+		if(!valid) return;
+		$aggregateSettings = $aggregateSettings.filter(a => JSON.stringify(a) !== JSON.stringify(oldAggregate));
+		$aggregateSettings = [...$aggregateSettings, newAggregate];
+		closeDialog();
 	}
 
-	const canWriteSchemaGroup = () => {
-		return currentId == 0 || (schemaGroup == undefined && schemaGroup.length == 0);
+	const closeDialog = () => {
```

Comment:
called after every "closing" action, it also resets local storage so you don't end up in the Dialog when you reload on the main page (the one with the color cards)